src/playback.jl

```diff
@@ -0,0 +1,45 @@
+
+"""
+    PlaybackPolicy{A, P<:Policy}
+a generic policy that uses a fixed sequence of actions until they are all used and then falls back onto a backup policy until the end of the episode
+
+Constructor:
+
+    `PlaybackPolicy(actions::AbstractArray, backup_policy::Policy; logpdfs::AbstractArray{Float64, 1} = Float64[])`
+
+# Fields
+- `actions::Vector{A}` a vector of actions to play back
+- `backup_policy::Policy` the policy to use when all prescribed actions have been taken but the episode continues
+- `logpdfs::Vector{Float64}` the log probability (density) of actions
+- `i::Int64` the current action index
+"""
+mutable struct PlaybackPolicy{A, P<:Policy, V<:Real} <: Policy
+    actions::AbstractArray{A}
+    backup_policy::P
+    logpdfs::AbstractArray{V}
```

Comment:
sorry, my previous comment was a bit unclear, if you use an abstract type it should be parameterized: 

```julia 
mutable struct PlaybackPolicy{A<:AbstractArray, P<:Policy, V<:AbstractArray{<:Real}} <: Policy
    actions::A
    backup_policy::P
    logpdfs::V
    i::Int64
end
```

Reply:
Thanks for pointing this out. I was unclear about the distinction between concrete and abstract types.